stustapay/core/service/order.py

```diff
@@ -71,34 +80,43 @@ async def create_order(
             customer_account.id,
         )
 
-        count = 0
+        lineitem_count = 0
         restricted_products = set()
         for item in new_order.positions:
-            item_id = item.product_id
+            product_id = item.product_id
             item_quantity = item.quantity
+            item_price = item.price
 
             # check product cost
             cost = await conn.fetchrow(
                 "select "
                 "    product.price, "
+                "    product.fixed_price, "
                 "    tax.rate, "
                 "    tax.name "
                 "from product "
                 "    left join tax on (tax.name = product.tax_name) "
                 "where product.id = $1;",
-                item_id,
+                product_id,
             )
             if cost is None:
-                raise NotFoundException(element_typ="product", element_id=str(item_id))
-            price, tax_rate, tax_name = cost
+                raise NotFoundException(element_typ="product", element_id=str(product_id))
+            price, fixed_price, tax_rate, tax_name = cost
+            if fixed_price and item_price:
+                raise InvalidArgumentException("The line item price was set for a fixed price item")
+            # other case (not fixed_price and not item_price) is implicitly checked with the database constraints,
```

Comment:
we can check this directly as constraint?

Reply:
At some point we need to check if the provided product has a fixed price or not. We cannot do this in the `NewLineItem` validator, as we would need to query the database for this.